tests/components/filter_menu.test.ts

```diff
@@ -96,6 +96,40 @@ describe("Filter menu component", () => {
       expect(values.map((val) => val.value)).toEqual(["2", "10", "a", "B"]);
     });
 
+    test("Dates are sorted in chronological order and non-date values are appended at the end", async () => {
+      createFilter(model, "A1:A5");
+      setCellContent(model, "A2", "September 2022");
+      setCellContent(model, "A3", "April 2023");
+      setCellContent(model, "A4", "Order Date (Month)");
+      setCellContent(model, "A5", "Jan 2023");
+
+      await openFilterMenu();
+      const menuValues = getFilterMenuValues();
+      expect(menuValues.map((val) => val.value)).toEqual([
+        "September 2022",
+        "Jan 2023",
+        "April 2023",
+        "Order Date (Month)",
+      ]);
+    });
+
+    test("[Intentional] Dates in other languages (except 'english') are not sorted in chronological order", async () => {
+      createFilter(model, "A1:A5");
+      setCellContent(model, "A2", "Şubat 2023");
+      setCellContent(model, "A3", "Temmuz 2022");
+      setCellContent(model, "A4", "Nisan 2023");
+      setCellContent(model, "A5", "Ekim 2022");
+
+      await openFilterMenu();
+      const menuValues = getFilterMenuValues();
+      expect(menuValues.map((val) => val.value)).toEqual([
```

Comment:
This test was useful :smile: 
I was wondering why the values were not ordered chronologically.
It turns out `Date.parse("sqfqsfdqsfqsdf 2020")` is happily parsed to the 01/01/2020 :man_facepalming: 
Hence it's considered a date instead of a regular string.

It's illustrated by this test I just wrote. It's passing in master but not with this branch
```ts
    test("sort according to letters before numbers", async () => {
      createFilter(model, "A1:A5");
      setCellContent(model, "A2", "AAA 2022");
      setCellContent(model, "A3", "BBB 2020");
      setCellContent(model, "A4", "CCC 2020");
      setCellContent(model, "A5", "DDD 2022");

      await openFilterMenu();
      const menuValues = getFilterMenuValues();
      expect(menuValues.map((val) => val.value)).toEqual([
        "AAA 2022",
        "BBB 2020",
        "CCC 2020",
        "DDD 2022",
      ]);
    });
```

Reply:
> this is very much harder than I thought. It doesn't even work properly for English without other side effects. If you can find an alternative strategy, we can explore it. Otherwise, we'll cancel the task

Yeah, I'm just wondering about that How “AAA 2022” will be converted to 1 January 2022.

Well, When I try to make it work for English dates only it also affects non-english dates.

I have made several attempts to address this issue, but the methods I have tried so far have proven to be both inflexible and expensive. I also attempted to utilize external libraries, such as Luxon, but found the implementation process to be overly complex. Additionally, Luxon requires a locale to function properly. Due to these challenges, I have not yet pushed the code. May I ask for your suggestion ?

Reply:
> Yeah, I'm just wondering about that How “AAA 2022” will be converted to 1 January 2022.

The magic of javascipt `Date` :man_shrugging: 

> I also attempted to utilize external libraries, such as Luxon

We will have to introduce Luxon in the near future for another task (support local date formats). Maybe we will be able to come up with something when we will do it.

> May I ask for your suggestion ?

I'd cancel the task

Reply:
Alright, make sense, Thanks for your inputs